### PR TITLE
Introduction to the WebDAV Protocol

### PR DESCRIPTION
**Pull request by jwodder**
Part of #148.

### CODE REVIEW

**Comment by jwodder** [comment]
FYI @yarikoptic.  You may be interested in this.

File: doc/webdav-intro.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+The `Dav` Header
+----------------
+
+A response to an `OPTIONS` request to a WebDAV-enabled URL must include a "Dav"
+header whose value is a comma-separated list of tokens identifying the
+"compliance classes" that the resource supports.  Class "1" is for resources
+that meet all "MUST" requirements in the RFC.  Class "2" is for resources that
+support class "1" and also various locking-related functionality.  Class "3" is
+for resources that support class "1" and also the changes made to RFC 2518 by
+RFC 4918.
```
Comment (yarikoptic) [inline doc/webdav-intro.md:214]: Is there some WebDAV validator/tester , to check WebDAV resource on compliance? Not unlike 'git annex testremote' testing compliance to annex external protocol?

File: doc/webdav-intro.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+XML Extensibility
+-----------------
+
+Custom properties and other elements can be added to WebDAV request & response
+bodies by placing them in an XML namespace other than "`DAV:`" (which is
+reserved for elements defined by RFCs).  If a server or client receives a
+document containing elements it does not recognize, it must ignore those
+elements and all of their children, except when a client attempts to set one or
+more unrecognized dead properties on a resource, in which case all of the
+properties must be recorded.
```
Comment (yarikoptic) [inline doc/webdav-intro.md:276]: Are there any good examples of such extensions?

File: doc/webdav-intro.md (hunk 1 above)
Comment (jwodder) [reply doc/webdav-intro.md:214]: I'm not aware of one, and I don't see one on http://webdav.org/projects/.

File: doc/webdav-intro.md (hunk 2 above)
Comment (jwodder) [reply doc/webdav-intro.md:276]: There are a number of RFCs that extend WebDAV (listed on the left of http://webdav.org under "Specifications"), though I expect they all use the `DAV:` namespace.

**Pull request merged**